selfhost/typechecker.jakt

```diff
@@ -3957,6 +3957,63 @@ struct Typechecker {
             let type_id = .find_or_add_type_id(Type::GenericInstance(id: array_struct_id, args: [inner_type_id]))
             yield CheckedExpression::JaktArray(vals, repeat, span, type_id)
         }
+        JaktDictionary(values, span) => {
+            let VOID_TYPE_ID = builtin(BuiltinType::Void)
+            mut key_type_id = unknown_type_id()
+            mut value_type_id = unknown_type_id()
+            mut vals: [(CheckedExpression, CheckedExpression)] = []
+            let dictionary_struct_id = .find_struct_in_prelude("Dictionary")
+
```

Comment:
Maybe add a // TODO for the hint parts that you didn't take over from rust. (also see my implementation as a suggestion)